src/main/java/com/google/devtools/build/lib/sandbox/LinuxSandboxedSpawnRunner.java

```diff
@@ -315,35 +337,36 @@ protected ImmutableSet<Path> getWritableDirs(Path sandboxExecRoot, Map<String, S
     return writableDirs.build();
   }
 
-  private SortedMap<Path, Path> getBindMounts(
-      BlazeDirectories blazeDirs, Path sandboxExecRoot, @Nullable Path sandboxTmp)
+  private Map<Path, Path> getBindMounts(
+      BlazeDirectories blazeDirs, SandboxInputs inputs, PathFragment sandboxSourceRoots,
+      Path sandboxExecRoot, @Nullable Path sandboxTmp)
       throws UserExecException {
     Path tmpPath = fileSystem.getPath("/tmp");
-    final SortedMap<Path, Path> bindMounts = Maps.newTreeMap();
```

Comment:
I think this will break stuff. I made this a SortedMap (also in the method parameters, so that people don't accidentally pass a non-sorted map in) when I wrote this code, because you need to ensure that you always mount parent directories before mounting anything below them.

Consider this case: `bazel build --sandbox_add_mount_pair=/foobar:/foo/bar --sandbox_add_mount_pair=/foo:/foo`

You need to first mount /foo on /foo and then /foobar on /foo/bar, if you do it the other way round, it will either fail (if the code doesn't create the necessary parent directory), or hide the other mount point accidentally.

depending on the values of the various flags adding directories to this map, it's possible that you end up in a non-working state.

Reply:
Yeah, that seems to be the least amount of "rocking the boat"; I have barely enough capacity to fix this narrow bug, let alone thinking deeply about what the exact behavior should be because as you say, it's tricky, both the semantics and the migration path.

So I'll go with the special-casing these mounts and keeping the `SortedMap`.

Reply:
Done.